src/TestProblem/Hydro/CCSN/Flu_ResetByUser_CCSN.cpp

```diff
@@ -0,0 +1,110 @@
+#include "GAMER.h"
+
+#if ( MODEL == HYDRO  &&  EOS == EOS_NUCLEAR  &&  NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
+
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Flu_ResetByUser_CCSN
+// Description :  Function to reset the temperature initial guess TEMP_IG in the CCSN problem
+//
+// Note        :  1. Invoked by Flu_ResetByUser_API_CCSN() and Model_Init_ByFunction_AssignData() using the
+//                   function pointer "Flu_ResetByUser_Func_Ptr", which must be set by a test problem initializer
+//                2. This function will be invoked when constructing the initial condition in
+//                   Model_Init_ByFunction_AssignData() and after each update in Flu_ResetByUser_API_CCSN()
+//                3. Input fluid[] stores the original values
+//                4. Even when DUAL_ENERGY is adopted, one does NOT need to set the dual-energy variable here
+//                   --> It will be set automatically in Flu_ResetByUser_API_CCSN() and
+//                       Model_Init_ByFunction_AssignData()
+//                5. Enabled by the runtime option "OPT__RESET_FLUID"
+//
+// Parameter   :  fluid    : Fluid array storing both the input (original) and reset values
+//                           --> Including both active and passive variables
+//                x/y/z    : Target physical coordinates
+//                Time     : Target physical time
+//                dt       : Time interval to advance solution
+//                lv       : Target refinement level
+//                AuxArray : Auxiliary array
+//
+// Return      :  true  : This cell has been reset
+//                false : This cell has not been reset
+//-------------------------------------------------------------------------------------------------------
+bool Flu_ResetByUser_Func_CCSN( real fluid[], const double x, const double y, const double z, const double Time,
+                                const double dt, const int lv, double AuxArray[] )
+{
+
+   if ( AuxArray != NULL ) {
```

Comment:
- Why use `AuxArray` to determine whether this cell is reset or not? Should we use the returned temperature for the condition?
- In `Flu_ResetByUser_API_CCSN()`, the first element of input `AuxArray` is already set to the correct value, `Emag` or `NULL_REAL`. Why do we to re-assign the value based on `MHD`?

Reply:
- This is due to `AuxArray` in `Hydro_Init_ByFunction_AssignData`. I will anyway remove it once new PR for `Emag` in `Flu_ResetByUser_Func_*`

- You are right. We do not need to do this twice. I removed it.